parser/Parser/Block.h

```diff
@@ -1,8 +1,16 @@
 #pragma once
-class Block
+#include <cstdint>
+#include "Reader.h"
+
+// not virtual! This might be counter-intuitive
+struct Block
```

Comment:
This, I did not intend to push, sry

```suggestion
class Block
```